tasks/post-to-cmr/index.js

```diff
@@ -198,34 +198,43 @@ async function moveGranuleFiles(granulesObject, sourceBucket) {
 
   Object.keys(granulesObject).forEach((granuleKey) => {
     const granule = granulesObject[granuleKey];
-    granule.files.forEach((file) => {
-      const source = {
-        Bucket: sourceBucket,
-        Key: `${file.fileStagingDir}/${file.name}`
-      };
-
-      const target = {
-        Bucket: file.bucket,
-        Key: file.filepath
-      };
+    const expectedFormat = /.*\.cmr\.xml$/;
 
-      moveFileRequests.push(moveGranuleFile(source, target));
+    granule.files.forEach((file) => {
+      if (!(file.name.match(expectedFormat))) {
+        log.info(file);
+        const source = {
+          Bucket: sourceBucket,
+          Key: `${file.fileStagingDir}/${file.name}`
+        };
+
+        log.info(`source key: ${source.Key} and bucket ${source.Bucket}`);
+
+        const target = {
+          Bucket: file.bucket,
+          Key: file.filepath
+        };
+        moveFileRequests.push(moveGranuleFile(source, target));
+      }
     });
   });
 
   return Promise.all(moveFileRequests);
 }
 
 /**
-* Update the granule metadata with the final location of files
+* Update the granule metadata with the final location of files.
+* For each granule file, find the collection regex that goes with it and use
+* that to construct the url path. Return the updated granules object.
 *
 * @param {Object} granulesObject - an object of the granules where the key is the granuleId
 * @param {Object} collection - configuration object defining a collection
 * of granules and their files
 * @param {string} cmrFiles - array of objects that include CMR xmls uris and granuleIds
+* @param {Array} buckets - the buckets involved with the files
 * @returns {Promise} promise resolves when all files have been moved
 **/
-function updateGranuleMetadata(granulesObject, collection, cmrFiles) {
+function updateGranuleMetadata(granulesObject, collection, cmrFiles, buckets) {
```

Comment:
the fileConfig.bucket (from collection config) has actual bucket name now, not the bucket type.
https://github.com/cumulus-nasa/cumulus/blob/cumulus-456-post-to-cmr/tasks/post-to-cmr/tests/data/payload.json#L57

Reply:
I'm confused about whether `fileConfig.bucket` should be the actual bucket name or if it should be the type (a key in the `buckets` object).

It seems like it shouldn't be the actual bucket name, but I could be missing something.

Reply:
Looking at it some more it seems like that test data should be revised to use the bucket types.

Reply:
I looked at sync granule from before seth moved it and it was a key in the buckets object. But we also have the ticket for multiple public and protected buckets coming it so this will have to change anyway